### ISSUE TITLE
Monolithic organization of contact module

### ISSUE THREAD

**Issue by lindsayad** [idaholab/moose#11617]
## Rationale

All contact headers and src are in a single directory.

## Description

We're not taking advantage of the unity build system by having all classes in the single base directory. It's also cleaner to parse the classes when they're organized into sub-directories.

## Impact

Current organization structure doesn't capitalize on the unity build system and it's hard to parse the current (lack of) directory structure

**Comment by permcody** [comment]
Closed by #14207
